pkg/scalers/azure_log_analytics_scaler.go

```diff
@@ -0,0 +1,492 @@
+		if err != nil {
+			return nil, fmt.Errorf("can't parse threshold: %s", err)
+		}
+		meta.threshold = threshold
+	} else {
+		return nil, fmt.Errorf("threshold was not found in metadata. Check your ScaledObject configuration")
+	}
+
+	return &meta, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *azureLogAnalyticsScaler) IsActive(ctx context.Context) (bool, error) {
+	err := s.updateCache()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+		return false, err
+	}
+
+	return s.cache.metricValue > 0, nil
+}
+
+func (s *azureLogAnalyticsScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	err := s.updateCache()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+		return nil
+	}
+
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: fmt.Sprintf("%s-%s", "azure-log-analytics", s.metadata.workspaceID),
+		},
+		Target: v2beta2.MetricTarget{
+			Type:         v2beta2.AverageValueMetricType,
+			AverageValue: resource.NewQuantity(s.cache.metricThreshold, resource.DecimalSI),
+		},
+	}
+	metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: externalMetricType}
+	return []v2beta2.MetricSpec{metricSpec}
+}
+
+//GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *azureLogAnalyticsScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	receivedMetric, err := s.getMetricData()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Failed to get metrics.")
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(receivedMetric.value, resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *azureLogAnalyticsScaler) Close() error {
+	return nil
+}
+
+func (s *azureLogAnalyticsScaler) updateCache() error {
+	if s.cache.metricValue < 0 {
+		receivedMetric, err := s.getMetricData()
+
+		if err != nil {
+			logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+			return err
+		}
+
+		s.cache.metricValue = receivedMetric.value
+
+		if receivedMetric.threshold > 0 {
+			s.cache.metricThreshold = receivedMetric.threshold
+		} else {
+			s.cache.metricThreshold = s.metadata.threshold
+		}
+	}
+
+	return nil
+}
+
+func (s *azureLogAnalyticsScaler) getMetricData() (metricsData, error) {
+	tokenInfo, err := s.getAccessToken()
+	if err != nil {
+		return metricsData{}, err
+	}
+
+	metricsInfo, err := s.executeQuery(s.metadata.query, tokenInfo)
+	if err != nil {
+		return metricsData{}, err
+	}
+
+	logAnalyticsLog.Info(fmt.Sprintf("Providing metric value: %d", metricsInfo.value))
```

Comment:
We are using [Structured logging](https://www.client9.com/structured-logging-in-golang/).

Would be better to replace the call to something like: 
logAnalyticsLog.Info("Getting metric value, "metrics value", metricsInfo.value))

Reply:
If this is DEBUG log message, would be better to call it `logAnalyticsLog.V(1).Info("Getting metr...)`
So we don't flood output with this particular log

Reply:
Thanks for pointing. I have changed this definition in last commit.